docs/sources/datasources/loki.md

```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+| `Legend`           | Available only in Dashboard. Controls the name of the time series, using name or pattern. For example `{{hostname}}` is replaced with the label value for the label `hostname`.                             |
 
-The following label matching operators are currently supported:
+### Log browser
 
-- `=` exactly equal.
-- `!=` not equal.
-- `=~` regex-match.
-- `!~` do not regex-match.
+// TODO David
 
-Examples:
+## Querying with Loki
 
-- `{name=~"mysql.+"}`
-- `{name!~"mysql.+"}`
+There are two types of LogQL queries:
 
-The [same rules that apply for Prometheus Label Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) apply for Loki Log Stream Selectors.
+- **Log queries** that return the contents of log lines.
+- **Metric queries** that extend log queries and calculate sample values based on the content of logs from a log query.
 
-Another way to add a label selector is in the table section. Click **Filter** beside a label to add the label to the query expression. This even works for multiple queries and will add the label selector to each query.
-
-### Search Expression
-
-After writing the Log Stream Selector, you can filter the results further by writing a search expression. The search expression can be just text or a regex expression.
-
-Example queries:
-
-- `{job="mysql"} |= "error"`
-- `{name="kafka"} |~ "tsdb-ops.*io:2003"`
-- `{instance=~"kafka-[23]",name="kafka"} != "kafka.server:type=ReplicaManager"`
-
-Filter operators can be chained and will sequentially filter down the expression. The resulting log lines will satisfy every filter.
+### Log Queries
 
-**Example**
+Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/#log-queries) query to display your logs.
 
-`{job="mysql"} |= "error" != "timeout"`
+A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons you need to start by choosing a log stream by selecting a log label.
 
-The following filter types are currently supported:
+### Log Context
 
-- `|=` line contains string.
-- `!=` line doesn't contain string.
-- `|~` line matches regular expression.
-- `!~` line does not match regular expression.
+When using a search expression as detailed above, you now have the ability to retrieve the context surrounding your filtered results.
+By clicking the `Show Context` link on the filtered rows, you'll be able to investigate the log messages that came before and after the
+log message you're interested in.
 
-> **Note:** For more details about LogQL, Loki's query language, refer to the [Loki LogQL](https://grafana.com/docs/loki/latest/logql/)
+> **Note:** This feature is only available in Grafana v6.3+
```

Comment:
Can we remove this note? 6.3 is really old.

Reply:
removed